pkg/tracing/populate_span_transformer.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing
+
+import (
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
+	"github.com/cloudevents/sdk-go/v2/types"
+	"go.opencensus.io/trace"
+	"knative.dev/eventing/pkg/kncloudevents"
+)
+
+func PopulateSpan(span *trace.Span, destination string) binding.TransformerFunc {
+	return func(reader binding.MessageMetadataReader, writer binding.MessageMetadataWriter) error {
+
+		span.AddAttributes(MessagingProtocolHTTP)
+		span.AddAttributes(MessagingSystemAttribute)
+
+		_, id := reader.GetAttribute(spec.ID)
+		if id != nil {
+			idParsed, err := types.Format(id)
+			if err != nil {
+				return err
+			}
+			span.AddAttributes(trace.StringAttribute(MessagingMessageIDAttributeName, idParsed))
+		}
+
+		span.AddAttributes(trace.StringAttribute(MessagingDestinationAttributeName, destination))
+
+		cloudeventsPopulator := kncloudevents.PopulateSpan(span)
```

Comment:
Why that? can't you just inline the code of `kncloudevents.PopulateSpan` here?

Reply:
If this is inlined, would you suggest to remove the code in `kncloudevents`?

Reply:
Yes, there's no need to have 2 transformers that sets the span attributes

Reply:
sure, I can inline it and get rid if the kncloudevents transformer